errs/error.go

```diff
@@ -33,11 +33,24 @@ func WithMessage(format string, args ...interface{}) Option {
 	}
 }
 
+// WithCtxKeyVal returns an Option that adds the given key-value pair to the
+// Error details. This is helpful for debugging errors.
+func WithCtxKeyVal(key string, val interface{}) Option {
```

Comment:
I would rename this, to `WithKeyValue`, after looking first at the method name, I thought you were using a `context.Context`